### DISCUSSION TITLE
Does Signac's LinkPeaks function compute correlations using single cells or using metacells?

### DISCUSSION THREAD

**Discussion by RegnerM2015** [stuart-lab/signac#1281] [Q&A] [answered]
Hi @timoast and others,

Thank you for maintaining an incredible resource! I am reaching out to improve my understanding of how Signac works in comparison to other methods. 

Based on the source code for LinkPeaks, it appears that the peak-to-gene correlations are computed using single cell observations as opposed to metacells or aggregates of similar cells. 

To my understanding, metacells are often created for these correlation analyses to circumvent the sparsity issues associated with scATAC-seq data. 

From ArchR documentation: https://www.archrproject.com/bookdown/creating-low-overlapping-aggregates-of-cells.html

"ArchR facilitates many integrative analyses that involve correlation of features. Performing these calculations with sparse single-cell data can lead to substantial noise in these correlative analyses. To circumvent this challenge, we adopted an approach introduced by [Cicero](https://cole-trapnell-lab.github.io/cicero-release/) to create low-overlapping aggregates of single cells prior to these analyses. We filter aggregates with greater than 80% overlap with any other aggregate in order to reduce bias. To improve the speed of this approach, we developed an implementation of an optimized iterative overlap checking routine and a implementation of fast feature correlations in C++ using the “Rcpp” package. These optimized methods are used in ArchR for calculating peak co-accessibility, peak-to-gene linkage, and for other linkage analyses. The use of these low-overlapping aggregates happens under the hood but we mention it here for clarity."

From Pliner et al. (Cicero): https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6582963/
"Grouping cells
In principle, Cicero could analyze the sample covariance computed between the vectors xi and xj of binary values encoding accessibility across cells for a pair of sites i and j. However, rather than working with the binary data directly, Cicero groups similar cells and aggregates their binary accessibility profiles into integer count vectors that are easier to work with in downstream steps. Under the grouping discussed below, the number of cells in which a particular site is accessible can be modeled with a binomial distribution or, for sufficiently large groups, the corresponding Gaussian approximation. Modeling grouped accessibility counts as normally distributed allows Cicero to easily adjust them for arbitrary technical covariates by simply fitting a linear model and taking the residuals with respect to it as the adjusted accessibility score for each group of cells."

There are likely pros and cons to both approaches (metacells v. single-cells). For metacells, you avoid sparsity but the tradeoff is that you lose independence of observations which may bias/inflate correlation values in some cases. For single cells, you increase N but sparsity may become problematic in some cases. Of course, these challenges would likely be dataset-depedent and depend on other technical factors like sequencing depth. 

Are there any other factors to consider here while comparing these approaches (metacells versus single cells)? 

Thank you for your help and time!

**Comment by timoast** [comment, marked as answer]
> Based on the source code for LinkPeaks, it appears that the peak-to-gene correlations are computed using single cell observations as opposed to metacells or aggregates of similar cells.

Yes, this is correct. We implemented the method described by SHARE-seq (Sai Ma et al. 2020)

It sounds like your question you're really asking is _should_ we use single cells or metacells, rather than what does the LinkPeaks function use currently?

I'm sure that the peak-gene linkage method that's use in the current LinkPeaks function can be improved, it has just not been priority of mine so far to extensively explore different methods for doing it and the method in SHARE-seq seemed to work well. Using metacells does seem to me like a good approach, and there are downsides to the peak-based bootstrapping approach used by the SHARE-seq method (eg, https://www.biorxiv.org/content/10.1101/2022.09.15.507748v1).

Long-term, I'd like to have several different peak-gene linkage methods implemented in Signac and users would then be able to easily run different methods, compare them, and choose the approach they think is most appropriate.

**Reply by RegnerM2015** [reply]
Thanks @timoast!

I appreciate your insights and feedback. This helps a lot in my efforts to digest the literature and the different approaches for finding peak-to-gene links. 

While my current work involves the ArchR toolkit, I would be happy to submit a pull request in the future to help increase access to different methods.
